### PR TITLE
adding code on progress bar

### PR DESCRIPTION
**Pull request by fabricio-aguiar**
ref #5184
https://pulp.plan.io/issues/5184
Required PR: https://github.com/pulp/pulpcore/pull/289
Required PR: https://github.com/pulp/pulp_file/pull/272

Please be sure you have read our documentation on creating PRs:
https://docs.pulpproject.org/en/3.0/nightly/contributing/pull-request-walkthrough.html

### CODE REVIEW

**Comment by goosemania** [comment]
Does it make sense to create a pre-defined set of codes which will be likely used by plugins as well? The goal is to make it easier to have at least some consistency in the code names across plugins. E.g. sync, publish, parsing metadata, etc.

**Comment by fabricio-aguiar** [comment]
> Does it make sense to create a pre-defined set of codes which will be likely used by plugins as well? The goal is to make it easier to have at least some consistency in the code names across plugins. E.g. sync, publish, parsing metadata, etc.

@goosemania I believe it makes sense, but  I don't know what would be these set of codes, and in some cases, it is hard to define what would be the code to differentiate the progress bars, like in this case: https://github.com/pulp/pulp_rpm/pull/1437/files

File: CHANGES/5184.misc [hunk 1]
```diff
@@ -0,0 +1 @@
+Adding `code` on `ProgressBar`.
```
Comment (daviddavis) [inline CHANGES/5184.misc]: This needs to probably be a feature.

File: pulpcore/plugin/stages/association_stages.py [hunk 2]
```diff
@@ -78,7 +78,7 @@ def __init__(self, new_version, *args, **kwargs):
         Returns:
             The coroutine for this stage.
         """
-        with ProgressBar(message='Un-Associating Content') as pb:
+        with ProgressBar(message='Un-Associating Content', code='un-associating.content') as pb:
```
Comment (daviddavis) [inline pulpcore/plugin/stages/association_stages.py]: s/un-associating/unassociating/

**Pull request merged**